### PR TITLE
23 events processing

### PR DESCRIPTION
**Pull request by mrlomar**
On status change, only the involved rules will be checked. Then if the conditions of a rule are evaluated to true, the actions of that rule are performed, and all related rules (rules with a condition about that rule) are also checked.

### CODE REVIEW

File: back-end/resources/room_config.json [hunk 1]
```diff
@@ -161,8 +161,8 @@
       "rules": [
         {
           "id": "controlSwitch",
-          "description": "Als de grote switch wordt geflipt, worden alle condities gecheckt (tweede schuif en groene of rode switch aan) en gaat het lichtje branden en de telefoon ringt.",
-          "limit": 1,
+          "description": "Wanneer slider 1 en 2 op 100 staan, en 3 oon 0 en de mainSwitch word geflipt, dan gaan alle groene lampjes aan",
```
Comment (IssaHanou) [inline back-end/resources/room_config.json]: oon -> op

File: back-end/resources/room_config.json [hunk 2]
```diff
@@ -175,48 +175,44 @@
...
-                      "comparison": "lt",
-                      "value": 30
+                      "component_id": "slider3",
+                      "comparison": "eq",
+                      "value": 0
                     }
                   ]
                 }
+              }, {
+                "type": "rule",
```
Comment (IssaHanou) [inline back-end/resources/room_config.json]: let's not make it dependent on the telephone

File: back-end/resources/room_manual.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ This is the general information of the escape room. It includes the following ta
 ### Devices
 This will be a list of all devices in the room. Each device is defined as a JSON object with the following properties:
 
-- `id`: this is the id of a device. Write it in camelCase, e.g. "controlBoard".
+- `id`: this is the id of a device. Write it in camelCase, e.g. "controlBoard".This id should be unique compared to other device ids and also the rule ids as well as the timer ids.
```
Comment (IssaHanou) [inline back-end/resources/room_manual.md]: extra space between sentences

File: back-end/resources/room_config.json [hunk 4]
```diff
@@ -253,79 +234,5 @@
       ]
     }
   ],
-  "general_events": [
```
Comment (IssaHanou) [inline back-end/resources/room_config.json]: Let's keep them in and change them next week when we will implement the front-end buttons

File: back-end/src/sciler/config/configHandler.go [hunk 5]
```diff
@@ -255,54 +303,6 @@ func generateLogicalConstraint(constraints interface{}) LogicalConstraint {
...
-//				}
-//			}
-//		} else {
-//			return actions, errors.New("invalid type of action: " + a.Type)
-//		}
-//	}
-//	return actions, nil
-//}
-
 //TODO multiple errors?
```
Comment (IssaHanou) [inline back-end/src/sciler/config/configHandler.go]: these todo's can be removed

File: back-end/resources/room_config.json (hunk 1 above)
Comment (mrlomar) [reply back-end/resources/room_config.json]: fixed

File: back-end/resources/room_manual.md (hunk 3 above)
Comment (mrlomar) [reply back-end/resources/room_manual.md]: fixed

File: back-end/resources/room_config.json (hunk 4 above)
Comment (mrlomar) [reply back-end/resources/room_config.json]: Sure, I will write a nice demo configuration

File: back-end/src/sciler/config/configHandler.go (hunk 5 above)
Comment (mrlomar) [reply back-end/src/sciler/config/configHandler.go]: Done

File: back-end/resources/room_config.json (hunk 2 above)
Comment (mrlomar) [reply back-end/resources/room_config.json]: I will make a new configuration

File: back-end/src/sciler/config/workingConfigTypes.go [hunk 6]
```diff
@@ -2,83 +2,123 @@ package config
...
 	default:
 		panic(fmt.Sprintf("cannot compare on: %s", comparision))
 	}
 }
 
+func numericToFloat64(input interface{}) float64 {
+	switch input.(type) {
+	case float64:
+		return input.(float64)
+	case int:
```
Comment (IssaHanou) [inline back-end/src/sciler/config/workingConfigTypes.go]: add case for doubles

File: back-end/src/sciler/main.go [hunk 7]
```diff
@@ -35,14 +35,15 @@ func main() {
...
 	go communicator.Start(messageHandler.NewHandler)
 
 	for _, value := range messageHandler.Config.Devices {
 		messageHandler.SendStatus(value.ID)
 	}
+
 	// loop for now preventing app to exit
 	for {
-		time.Sleep(time.Microsecond * time.Duration(10))
+		time.Sleep(time.Nanosecond * time.Duration(10))
```
Comment (GwennanS) [inline back-end/src/sciler/main.go]: why?

**Pull request merged**